datalad/interface/utils.py

```diff
@@ -514,16 +515,24 @@ def default_result_renderer(res):
                 if res.get('message', None) else ''))
 
 
-def _display_suppressed_message(nsimilar, ndisplayed, final=False):
+def _display_suppressed_message(nsimilar, ndisplayed, last_ts, final=False):
     # +1 because there was the original result + nsimilar displayed.
     n_suppressed = nsimilar - ndisplayed + 1
     if n_suppressed > 0:
-        ui.message('  [{} similar {} been suppressed]'
-                   .format(n_suppressed,
-                           single_or_plural("message has",
-                                            "messages have",
-                                            n_suppressed, False)),
-                   cr="\n" if final else "\r")
+        ts = time()
+        # rate-limit update of suppression message, with a large number
+        # of fast-paced results updating for each one can result in more
+        # CPU load than the actual processing
+        # arbitrarily go for a 2Hz update frequency -- it "feels" good
+        if last_ts is None or (ts- last_ts > 0.5):
```

Comment:
```suggestion
        if last_ts is None or (ts - last_ts > 0.5):
```
my invaluable contribution ;-)